insta_reels_publishing_api_sample/index.js

```diff
@@ -0,0 +1,221 @@
+        const igResponse = await axios.get(getInstagramAccountUri);
+        const hasIgBusinessAccount = igResponse.data.instagram_business_account ? true : false;
+
+        // If there is a IG Business Account associated with the page
+        if(hasIgBusinessAccount) {
+            try {
+                const igUserId = igResponse.data.instagram_business_account.id;
+
+                // Upload Reel Video
+                const uploadVideoUri = `https://graph.facebook.com/v14.0/${igUserId}/media?media_type=REELS&video_url=${yourVideoUrl}&caption=${yourCaption}&access_token=${req.session.userToken}`;
+                const uploadResponse = await axios.post(uploadVideoUri);
+                const containerId = uploadResponse.data.id;
+
+                // add variables to the session
+                Object.assign(req.session, { igUserId, containerId, pageToken });
+
+                // Render Upload Success
+                res.render("upload_page", {
+                    uploaded: true,
+                    igUserId,
+                    containerId,
+                    pages: req.session.pageData,
+                    message: `Reel uploaded successfully on IG UserID #${igUserId} at Container ID #${containerId}. You can Publish now.`
+                });
+            } catch(e) {
+                res.render("upload_page", {
+                    uploaded: false,
+                    error: true,
+                    pages: req.session.pageData,
+                    message: `Error during upload. [Selected page id - ${selectedPageID}, IG User ID - ${igUserId}`
+                });
+            }
+        } else { // Error - No IG Account found
+            res.render("upload_page", {
+                uploaded: false,
+                error: true,
+                pages: req.session.pageData,
+                message: "No Instagram Business Account associated with the page!"
+            });
+        }
+    } catch(e) {
+        res.render("upload_page", {
+            uploaded: false,
+            error: true,
+            pages: req.session.pageData,
+            message: `Error in getting IG account details for the selected page id - ${selectedPageID}`
+        });
+    }
+
+});
+
+app.post("/publishReels", async function (req, res) {
+    const { igUserId, containerId, pageToken } = req.session;
+
+    // Upload happens asynchronously in the backend,
+    // so you need to check upload status before you Publish
+    const checkStatusUri = `https://graph.facebook.com/v14.0/${containerId}?fields=status_code&access_token=${req.session.userToken}`;
+    const isUploaded = await isUploadSuccessful(0, checkStatusUri);
+
+    // When uploaded successfully, publish the video
+    if(isUploaded) {
+        const publishVideoUri = `https://graph.facebook.com/v14.0/${igUserId}/media_publish?creation_id=${containerId}&access_token=${req.session.userToken}`;
+        const publishResponse = await axios.post(publishVideoUri);
```

Comment:
It is potentially worth considering having a try catch block around any axios post/get request, just to handle any potential exceptions.

Reply:
Yea it could be done, but I don't wanted to make the code more verbose, as this is just a sample app. And the developers who would be using it as reference, will be able to handle the exceptions the way they want to. What do you think @ani0989